bin/network-monitor/src/commands/start.rs

```diff
@@ -45,11 +47,23 @@ pub async fn start_monitor(config: MonitorConfig) -> Result<()> {
         None
     };
 
+    // Initialize the counter increment task only if enabled.
+    let counter_rx = if config.enable_counter {
+        // Ensure accounts exist before starting monitoring tasks
+        ensure_accounts_exist(&config.wallet_filepath, &config.counter_filepath, &config.rpc_url)
+            .await?;
```

Comment:
Unless the accounts are used for something else, I would put the account creation/loading within the service, only for a slightly more encapsulated organization.